### PR TITLE
Introduce config for default task duration

### PR DESCRIPTION
**Pull request by sublinus**
This introduces the key `distributed.scheduler.task-duration` in the Dask config to configure a default task duration for all tasks, when there is no otherwise configured value available.

That is useful in cases of very long-running tasks to tune the occupancy of the cluster (see #3516 and #3627 )

### CODE REVIEW

**Comment by quasiben** [comment]
Thanks @sublinus .  Looks reasonable to me and definitely appreciate the test!  I am going to wait until the end of the day to merge should other maintainers want to comment

File: distributed/distributed.yaml [hunk 1]
```diff
@@ -24,6 +24,9 @@ distributed:
     pickle: True            # Is the scheduler allowed to deserialize arbitrary bytestrings
     preload: []
     preload-argv: []
+    # Assume this duration ("15m", "2h") for all tasks with unknown durations
+    # This can be useful to change occupancy estimations in Adaptive scenarios
+    task-duration: null
```
Comment (mrocklin) [inline distributed/distributed.yaml]: Should this have a non-null value?  What do we do currently?

So far we've been limiting comments for configuration values to be a bit shorter and be on the same line.  If we choose to change convention here and have much larger comments then we should probably do that across the file.  For now I recommend that we try to make the comment more brief if we can.

**Comment by sublinus** [comment]
Thanks for all your comments. The question regarding the non-null value just pointed me to `scheduler.get_task_duration`, which is where I think this `unknown-task-duration` should be used instead of in the `TaskPrefix` as an average. 
Currently the default used for tasks with unknown task durations is 0.5s, which is now reflected in the config.
I totally see your point regarding the comment style.

**Comment by sublinus** [comment]
The partial CI failure seems to be unrelated here.

**Comment by quasiben** [comment]
@sublinus i think you are probably right but I restarted the tests job to be sure.

File: distributed/distributed.yaml [hunk 2]
```diff
@@ -24,6 +24,7 @@ distributed:
     pickle: True            # Is the scheduler allowed to deserialize arbitrary bytestrings
     preload: []
     preload-argv: []
+    unknown-task-duration: 0.5s  # Default duration for all tasks with unknown durations ("15m", "2h")
```
Comment (mrocklin) [inline distributed/distributed.yaml]:
```suggestion
    unknown-task-duration: 500ms  # Default duration for all tasks with unknown durations ("15m", "2h")
```

File: distributed/tests/test_scheduler.py [hunk 3]
```diff
@@ -2057,3 +2057,14 @@ def reducer(x, y):
...
     assert s.aliases == {a.name: a.address}
+
+
+@gen_cluster(client=True, config={"distributed.scheduler.unknown-task-duration": "1h"})
+async def test_unknown_task_duration_config(client, s, a, b):
+    future = client.submit(inc, 1)
+    while not s.tasks:
+        await asyncio.sleep(0.001)
+    assert sum(s.get_task_duration(ts) for ts in s.tasks.values()) == 3600
+    assert len(s.unknown_durations) == 1
```
Comment (mrocklin) [inline distributed/tests/test_scheduler.py]: What if `inc` finishes before we get here?

Comment (mrocklin) [reply distributed/tests/test_scheduler.py]: You might want to consider `slowinc`

Comment (sublinus) [reply distributed/tests/test_scheduler.py]: Good point. It should be easy to switch to `slowinc` here.

**Comment by sublinus** [comment]
The build shows as in progress even though GH Actions and Travis ran through fine. It looks like the notify about the successful build from Travis to Github failed.

**Comment by mrocklin** [comment]
Thanks @sublinus !  This is in

**Pull request merged**